classes/class3/README.md

```diff
@@ -1 +1,364 @@
-#Class 3
\ No newline at end of file
+#Class 3 - Express, Templating, and MongoDB
+
+##Express
+Now that we've explored Node.js a little, we will abstract the details away with the [Express](http://expressjs.com/) development framework. Before, in the Node Beginner's Book, our code looked a lot like this:
+```javascript
+var http = require("http");
+
+http.createServer(function(request, response) {
+  response.writeHead(200, {"Content-Type": "text/plain"});
+  response.write("Hello World");
+  response.end();
+}).listen(8888);
+```
+
+Now let's see what the same app would look like using Express. To start we'll set up a new application using `npm init`. This will ask you a series of questions that will populate your `package.json` file, you can use the defaults for all of them. This file manages many things. If you publish your module on npm (like any of the modules which you `npm install` are published) this file will include the meta information about your module. We won't get into that much in this class, but that's why it seems like there are a lot of unnecessary fields. The main purpose we will use the package.json for is dependency management.
+
+After you exit the init setup, run `npm install express --save`. If you check your package.json again, you'll see the following was added:
+```javascript
+"dependencies": {
+    "express": "^4.10.6"
+  },
+```
+The `--save` argument will add the module to your list of dependencies and is generally a good thing to use whenever adding a new module. If you just forked a cloned a project that has its dependencies listed you can simple run `npm install` to install the required dependencies (at the right versions) for that project.
+
+Now that we're set up, create a file named `app.js` and and paste in the following:
+```javascript
+var express = require('express');
+var app = express();
+
+app.get('/', function(req, res){
+  res.send('hello world');
+});
+
+app.listen(3000);
+```
+Inside the folder you created the `app.js` file, run the command `node app` to start the server. Go to `http://localhost:3000` in your web browser and you should see the following:
+```
+hello world
+```
+Awesome! You just ran your first application using Express!
+
+Express makes writing web servers in Node much easier. Here are some of the important differences between the examples you did in the book and Express:
+* Routing. Instead of trying to parse the URL the user is at ourselves, we can just tell Express to match an individual path with a function.
```

Comment:
* **Routing:**

or
* Routing:

but not
* Routing.

Same below.

Also, maybe change to "Instead of parsing the URL out of the request ourselves, we can have Express do that for us, and even assign a function to execute when a request asks for a specific path.